### PR TITLE
[Fuchsia] Include baremetal ARM builtins and libc

### PR DESCRIPTION
**Pull request by petrhosek**
For now we only include the armv6m-unknown-eabi target but we plan to include more targets in the future (including multilibs).

### CODE REVIEW

**Review by PiJoules — APPROVED** [review]

File: clang/cmake/caches/Fuchsia-stage2.cmake [hunk 1]
```diff
@@ -300,10 +300,43 @@ if(FUCHSIA_SDK)
...
+  set(RUNTIMES_${target}_LLVM_INCLUDE_TESTS OFF CACHE BOOL "")
+  set(RUNTIMES_${target}_LLVM_ENABLE_ASSERTIONS OFF CACHE BOOL "")
+  set(RUNTIMES_${target}_LLVM_ENABLE_RUNTIMES "libc" CACHE STRING "")
+endforeach()
+
 foreach(target riscv32-unknown-elf)
   list(APPEND BUILTIN_TARGETS "${target}")
   set(BUILTINS_${target}_CMAKE_SYSTEM_NAME Generic CACHE STRING "")
-  set(BUILTINS_${target}_CMAKE_SYSTEM_PROCESSOR RISCV CACHE STRING "")
+  set(BUILTINS_${target}_CMAKE_SYSTEM_PROCESSOR riscv CACHE STRING "")
```
Comment (ilovepi) [inline clang/cmake/caches/Fuchsia-stage2.cmake:339]: nit: does this and the riscv bit below need to be modified in this change? It seems unrelated.

**Review by ilovepi — APPROVED** [review]
LGTM other than the nit.

File: clang/cmake/caches/Fuchsia-stage2.cmake (hunk 1 above)
Comment (petrhosek) [reply clang/cmake/caches/Fuchsia-stage2.cmake]: Removed
